### PR TITLE
Make reference page of all values and variables available

### PR DESCRIPTION
**Pull request by rosieyohannan**
https://circleci.atlassian.net/browse/DOCSTEAM-20?atlOrigin=eyJpIjoiNjg1ZDE4MWRkOWRlNGEzMjhhMGI1M2VhZWNiODVkMGIiLCJwIjoiaiJ9

- [x] Put built-in env vars into reusable snippet
- [x] Put pipeline values into reusable snippet
- [x] Leave in place but also add to reference page for built-in values available to all projects
- [x] Pipelines team say pipelines variables is not the right term, so be explicit - values and parameters - to avoid any confusion

### CODE REVIEW

File: jekyll/_cci2/pipeline-variables.md [hunk 1]
```diff
@@ -1,17 +1,15 @@
 ---
 layout: classic-docs
-title: "Pipeline Variables"
-short-title: "Pipeline Variables"
-description: "Detailed information about Pipeline variables, parameters and values"
+title: "Pipeline values and variables"
```
Comment (rosieyohannan) [inline jekyll/_cci2/pipeline-variables.md:3]:
```suggestion
title: "Pipeline values and parameters"
```

File: jekyll/_includes/snippets/built-in-env-vars.adoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+| **Deprecated**. The GitHub or Bitbucket URL to compare commits of a build. Available in config v2 and below. For v2.1 we will introduce ["pipeline values"]({{site.baseurl}}/2.0/pipeline-variables/) as an alternative.
+
+| `CI_PULL_REQUEST`             
+| String  
+| **Deprecated**. Kept for backward compatibility with CircleCI 1.0. Use `CIRCLE_PULL_REQUEST` instead.
+
+| `CI_PULL_REQUESTS`             
+| List    
+| **Deprecated**. Kept for backward compatibility with CircleCI 1.0. Use `CIRCLE_PULL_REQUESTS` instead.
+|===
```
Comment (rosieyohannan) [inline jekyll/_includes/snippets/built-in-env-vars.adoc:115]: maybe we can remove these now?

File: jekyll/_includes/snippets/pipeline-values.adoc [hunk 3]
```diff
@@ -0,0 +1,55 @@
+[.table.table-striped]
+[cols=3*, options="header", stripes=even]
+|===
+| Variable
+| Type                      
+| Value
+
+| `pipeline.id`                 
+|
```
Comment (rosieyohannan) [inline jekyll/_includes/snippets/pipeline-values.adoc:9]: I need help with the types in this tables if possible

File: jekyll/_cci2/env-vars.md [hunk 4]
```diff
@@ -565,46 +565,11 @@ Start a run with the POST API call, see the [new build](https://circleci.com/doc
 ## Built-in environment variables
 {: #built-in-environment-variables }
 
-The following environment variables are exported in each build and can be used for more complex testing or deployment.
+Built-in environment variables are exported in each build and can be used for more complex testing or deployment.
```
Comment (sulami) [inline jekyll/_cci2/env-vars.md:568]:
```suggestion
Built-in environment variables are exported in each job and can be used for more complex testing or deployment. 
```
While we're at it.

File: jekyll/_cci2/pipeline-variables.md [hunk 5]
```diff
@@ -1,17 +1,15 @@
...
+- Server v3.x
 ---
 
-Pipeline variables can be used to create reusable pipeline configurations. To use pipeline variables you must have [pipelines enabled]({{ site.baseurl }}/2.0/build-processing) and use configuration [version]({{ site.baseurl }}/2.0/configuration-reference/#version) `2.1` or higher.
-
-There are two types of pipeline variables:
+Pipeline values and parameters can be used to create reusable pipeline configurations.
 
 * **Pipeline values** represent pipeline metadata that can be used throughout the configuration.
 * **Pipeline parameters** are typed pipeline variables that are declared in the `parameters` key at the top level of a configuration. Users can pass `parameters` into their pipelines when triggering a new run of a pipeline through the API.
```
Comment (sulami) [inline jekyll/_cci2/pipeline-variables.md:15]: Maybe not use the term pipeline variables in this place either?

File: jekyll/_includes/snippets/pipeline-values.adoc (hunk 3 above)
Comment (sulami) [reply jekyll/_includes/snippets/pipeline-values.adoc:9]: Pipeline number is an integer.

in_setup is a boolean.

The rest is strings of various formats (UUIDs, hashes, etc.).

File: jekyll/_cci2/variables.adoc [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+          password: $DOCKERHUB_PASSWORD  # context / project UI env-var reference
+    environment:
+      CIRCLE_COMPARE_URL: << pipeline.project.git_url >>/compare/<< pipeline.git.base_revision >>..<<pipeline.git.revision>>
+    working_directory: ~/main
+    steps:
+      - run: echo "This is pipeline ID << pipeline.id >>"
+      - run: echo $CIRCLE_COMPARE_URL
+```
+
+**Note:** When using the above method to set the variables in the `environment` key if the pipeline variable is empty it will set the variable to `<nil>`. If you need an empty string instead, [set the variable in a shell command]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command).
```
Comment (devinmarieb) [inline jekyll/_cci2/variables.adoc:52]: This is an awkward sentence. I would reword it slightly. Maybe "When using the above method to set the variables in the `environment` key, note that if the pipeline variable is empty it will set the variable to `<nil>`."

File: jekyll/_includes/snippets/built-in-env-vars.adoc (hunk 2 above)
Comment (devinmarieb) [reply jekyll/_includes/snippets/built-in-env-vars.adoc:115]: I would say yes, remove them? Not very confident in my opinion, but I do agree.

**Comment by devinmarieb** [comment]
@rosieyohannan I left two small comments. I didn't make a note of it, but also wondering if we should think about removing the specific "GitHub" and "Bitbucket" references here as well.

**Review by devinmarieb — APPROVED** [review]

File: jekyll/_includes/snippets/built-in-env-vars.adoc (hunk 2 above)
Comment (rosieyohannan) [reply jekyll/_includes/snippets/built-in-env-vars.adoc:115]: I'm going to do it! 🗑️

File: jekyll/_cci2/pipeline-variables.md [hunk 7]
```diff
@@ -57,7 +43,7 @@ jobs:
       - run: echo $CIRCLE_COMPARE_URL
 ```
 
-**Note:** When using the above method to set the variables in the `environment` key if the pipeline variable is empty it will set the variable to `<nil>`. If you need an empty string instead, [set the variable in a shell command]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command).
+**Note:** When using the above method to set the values in the `environment` key if the pipeline variable is empty it will set the variable to `<nil>`. If you need an empty string instead, [set the variable in a shell command]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command).
```
Comment (rosieyohannan) [inline jekyll/_cci2/pipeline-variables.md:46]:
```suggestion
**Note:** When using the above method to set the values in the `environment` key, note that if the pipeline variable is empty it will be set to `<nil>`. If you need an empty string instead, [set the variable in a shell command]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command).
```

File: jekyll/_cci2/variables.adoc (hunk 6 above)
Comment (rosieyohannan) [inline jekyll/_cci2/variables.adoc:52]:
```suggestion
**Note:** When using the above method to set the variables in the `environment` key, note that if the pipeline variable is empty it will be set to `<nil>`. If you need an empty string instead, [set the variable in a shell command]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command).
```

File: jekyll/_includes/snippets/built-in-env-vars.adoc [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+| String  
+| **Deprecated**. The GitHub or Bitbucket URL to compare commits of a build. Available in config v2 and below. For v2.1 we will introduce ["pipeline values"]({{site.baseurl}}/2.0/pipeline-variables/) as an alternative.
+
+| `CI_PULL_REQUEST`             
+| String  
+| **Deprecated**. Kept for backward compatibility with CircleCI 1.0. Use `CIRCLE_PULL_REQUEST` instead.
+
+| `CI_PULL_REQUESTS`             
+| List    
+| **Deprecated**. Kept for backward compatibility with CircleCI 1.0. Use `CIRCLE_PULL_REQUESTS` instead.
```
Comment (rosieyohannan) [inline jekyll/_includes/snippets/built-in-env-vars.adoc:114]:
```suggestion
```

**Comment by rosieyohannan** [comment]
I've realised that the snippets need to be in the same language as the page they are used in - I thought it didn't matter :-( - so for now I'll switch back to markdown for the snippet tables.

File: jekyll/_cci2/variables.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+version:
+- Cloud
+- Server v3.x
+- Server v2.x
+---
+
+This page is a reference for all built-in values available for use in your CircleCI projects.
+
+## Built-in environment variables
+
```
Comment (rosieyohannan) [inline jekyll/_cci2/variables.md:14]:
```suggestion
## Built-in environment variables
{: #built-in-environment-variables }

```

File: jekyll/_cci2/variables.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+The following built-in environment variables are available for all CircleCI projects. Environment variables are scoped at the job level. They can be used within the context of a job but do not exist at a pipeline level, therefore they cannot be used for any logic at the pipeline or workflow level.
+
+**Note**: You cannot use a built-in environment variable to define another environment variable. Instead, you must use a `run` step
+to export the new environment variables using `BASH_ENV`. For more details, see [Setting an Environment Variable in a Shell Command]({{ site.baseurl }}/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command).
+
+{% include snippets/built-in-env-vars.md %}
+
+## Pipeline values
+
```
Comment (rosieyohannan) [inline jekyll/_cci2/variables.md:23]:
```suggestion
## Pipeline values
{: #pipeline-values }

```
